enterprise/backend/src/metabase_enterprise/metabot_v3/tools/table_view_config.clj

```diff
@@ -0,0 +1,178 @@
+(ns metabase-enterprise.metabot-v3.tools.table-view-config
+  "Generate table view configurations using OpenAI API"
+  (:require
+   [cheshire.core :as json]
+   [clj-http.client :as http]
+   [clojure.string :as str]
+   [metabase.channel.template.core :as template]
+   [metabase.util :as u]
+   [metabase.util.i18n :as i18n]
+   [metabase.util.log :as log]
+   [toucan2.core :as t2]))
+
+(defn- get-openai-api-key
+  "Get OpenAI API key from environment variable"
+  []
+  (or (System/getenv "OPENAI_API_KEY")
+      (throw (ex-info (i18n/tru "OpenAI API key not configured")
+                      {:error-type :configuration-error}))))
+
+(defn- fetch-table-fields
+  "Fetch fields for a given table"
+  [table-id]
+  (let [fields (t2/select :model/Field
+                          :table_id table-id
+                          :active true
+                          {:order-by [[:position :asc]]})]
+    (map (fn [field]
+           {:id           (:id field)
+            :name         (:name field)
+            :display_name (:display_name field)
+            :type          (name (:base_type field))
+            :semantic_type (when (:semantic_type field)
+                             (name (:semantic_type field)))
+            :description   (:description field)})
+         fields)))
+
+(defn- build-prompt
+  "Build prompt for OpenAI based on view type and fields using handlebars template"
+  [view-type fields table-name]
+  (let [view-type-str (name view-type)
+        fields-json   (json/generate-string fields)
+        template-data {:viewType   view-type-str
+                       :tableName  table-name
+                       :fieldsJson fields-json}]
+    (template/render "metabase_enterprise/metabot_v3/tools/table_view_config_prompt.hbs" template-data)))
+
+(def ^:private response-schemas
+  "JSON schemas for structured output by view type"
+  {:listing
```

Comment:
Curious whether generating the schema with Malli would give sufficient precision, and how much it would help readability. I'm struggling even harder to read without alignment 😄